### DISCUSSION TITLE
Is working with quotients of associative algebras not implemented?

### DISCUSSION THREAD

**Discussion by mikepierce** [gap-system/gap#4247] [Q&A]
I'm trying to calculate the dimension of a quotient of a fingen unital associative algebra with GAP, but I'm running into an error that says the procedure depends on something not available yet.

    Error, this case will eventually be handled by the Vector Enumerator
    which is not available yet called from
    OperationAlgebraHomomorphism( A, [ [ Zero( A ) ] ], OnRight ) at /usr/share/gap/lib/algfp.gi:604 called from
    IsomorphismMatrixFLMLOR( A ) at /usr/share/gap/lib/algfp.gi:666 called from
    IsFiniteDimensional( V ) at /usr/share/gap/lib/modfree.gi:318 called from
    <function "unknown">( <arguments> )
     called from read-eval loop at line 7 of *stdin*
    you can 'quit;' to quit to outer loop, or
    you can 'return;' to continue

Is there a workaround for this? Or does this sort of calculation only work with Lie algebras in GAP? Thank you for your time! :) Here's a MWE to produce the error:

    Gens := ["a","b","c","d"];
    A := FreeAssociativeAlgebraWithOne(Rationals, Gens);
    GensA := GeneratorsOfAlgebra(A); 
    one := GensA[1]; a := GensA[2]; b := GensA[3]; c := GensA[4]; d := GensA[5];
    I := [ a*b-c*d, a*a-c ];
    AQuot := A/I;

    ### Any of these will return a similar error
    # LieAlgebra(AQuot);
    # NiceAlgebraMonomorphism(AQuot);
    # IsomorphismMatrixAlgebra(AQuot);
    # Dimension(AQuot);

**Comment by dimpase** [comment]
in fact GAP has had a vector enumeration package, written by @stevelinton - see
https://github.com/gap-packages/ve 

I did some work porting it to a modern version of GMP (these changes are already in that repo). It's probably not much work to make a proper port to a GAP 4 package.
